src/tequila/grouping/binary_rep.py

```diff
@@ -230,23 +232,41 @@ def anti_commutativity_matrix(self):
         gram = np.block([[np.zeros((n,n)), np.eye(n)], [np.eye(n), np.zeros((n,n))]])
         return matrix @ gram @ matrix.T % 2
 
-    def commuting_groups(self, method='rlf'):
+    def commuting_groups(self, method='rlf', condition=None, overlap_help=None):
         """
         Return the partitioning of the hamiltonian into commuting groups.
         List of BinaryHamiltonian's
         """
+        def method_class(method):
+            """
+            Return the class that the method belongs to: One from Minimum clique cover (mcc)
+            and Greedy grouping algorithms. 
+            """
+            if (method == 'lf' or method == 'rlf'): 
+                mc = 'mcc' 
+            elif (method == 'si' or method == 'osi'):
+                mc = 'greedy'
+            else:
+                raise TequilaException(f"There is no algorithm {method}")
+            return mc
         terms = self.binary_terms
         n = self.n_term
-        cg = self.anti_commutativity_matrix()
-
-        if method == 'lf':
-            colors = largest_first(terms, n, cg)
-        elif method == 'rlf':
-            colors = recursive_largest_first(terms, n, cg)
-        else:
-            raise TequilaException(f"There is no algorithm {method}")
-        return [BinaryHamiltonian(value) for key, value in colors.items()]
 
+        if method_class(method) == 'mcc':
+            cg = self.anti_commutativity_matrix()
+            if method == 'lf':
+                colors = largest_first(terms, n, cg)
+            elif method == 'rlf':
+                colors = recursive_largest_first(terms, n, cg)
+            return [BinaryHamiltonian(value) for key, value in colors.items()]
```

Comment:
Currently `mcc` methods always works as if `condition="fc"`, regardless of the actual value of `condition`. For example, `. commuting_groups('rlf', condition="qwc")` still gives FC groups instead of QWC groups. It'd be nice if the code prevents this.